### PR TITLE
feat: add `.root()` to avoid re-parsing inputs when possible

### PR DESCRIPTION
**Pull request by jquense**
## Description
Accept a postcss Root object as an input. Allows for minor optimization in a tool chain, where the input may also be using postcss...such as webpack loader chain, where the previous loader may return it's AST as well as the resulting string.

## Motivation and Context
nitty-gritty optimization territory. Cases where it doesn't make sense, or not preferable to pass plugins directly to `processor.before`.

## How Has This Been Tested?
I added a test!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/processor/processor.js [hunk 1]
```diff
@@ -351,7 +362,7 @@ class Processor {
 
     // Take a file id and some text, walk it for dependencies, then
     // process and return details
-    async _add(id, text) {
+    async _add(id, textOrRoot) {
```
Comment (tivac) [inline packages/processor/processor.js]: `textOrRoot` -> `src`

File: packages/processor/processor.js [hunk 2]
```diff
@@ -424,7 +435,7 @@ class Processor {
 
     // Process files and walk their composition/value dependency tree to find
     // new files we need to process
-    async _walk(name, text) {
+    async _walk(name, textOrRoot) {
```
Comment (tivac) [inline packages/processor/processor.js]: `textOrRoot` -> `src`

File: packages/processor/processor.js [hunk 3]
```diff
@@ -433,6 +444,10 @@ class Processor {
             return;
         }
 
+        const text = typeof textOrRoot === "string" ?
```
Comment (tivac) [inline packages/processor/processor.js]: Do this inside the assignment to `this._files[name]` since nothing else depends on it.

**Comment by tivac** [comment]
@jquense this is great and I'm kinda gobsmacked I didn't already add this 😳 Thanks!

**Comment by jquense** [comment]
thanks!

**Comment by tivac** [comment]
This is released in `@modular-css/processor@25.4.0`, and also [documented on m-css.com](https://m-css.com/api/#processorrootname-root)

**Pull request merged**